lib/snippets.js

```diff
@@ -0,0 +1,675 @@
+
+    this.userSnippetsPath = CSON.resolve(path.join(atom.getConfigDirPath(), 'snippets'))
+    if (this.userSnippetsPath == null) { this.userSnippetsPath = path.join(atom.getConfigDirPath(), 'snippets.cson') }
+    return this.userSnippetsPath
+  },
+
+  loadAll () {
+    this.loadBundledSnippets(bundledSnippets => {
+      this.loadPackageSnippets(packageSnippets => {
+        this.loadUserSnippets(userSnippets => {
+          atom.config.transact(() => {
+            for (let snippetSet of [bundledSnippets, packageSnippets, userSnippets]) {
+              for (let filepath in snippetSet) {
+                const snippetsBySelector = snippetSet[filepath]
+                this.add(filepath, snippetsBySelector)
+              }
+            }
+          })
+          this.doneLoading()
+        })
+      })
+    })
+  },
+
+  loadBundledSnippets (callback) {
+    const bundledSnippetsPath = CSON.resolve(path.join(getPackageRoot(), 'lib', 'snippets'))
+    this.loadSnippetsFile(bundledSnippetsPath, snippets => {
+      const snippetsByPath = {}
+      snippetsByPath[bundledSnippetsPath] = snippets
+      callback(snippetsByPath)
+    })
+  },
+
+  loadUserSnippets (callback) {
+    const userSnippetsPath = this.getUserSnippetsPath()
+    return fs.stat(userSnippetsPath, (error, stat) => {
+      if (stat != null && stat.isFile()) {
+        this.loadSnippetsFile(userSnippetsPath, snippets => {
+          const result = {}
+          result[userSnippetsPath] = snippets
+          callback(result)
+        })
+      } else {
+        callback({})
+      }
+    })
+  },
+
+  watchUserSnippets (callback) {
+    const userSnippetsPath = this.getUserSnippetsPath()
+    fs.stat(userSnippetsPath, (error, stat) => {
+      if (stat != null && stat.isFile()) {
+        const userSnippetsFileDisposable = new CompositeDisposable()
+        const userSnippetsFile = new File(userSnippetsPath)
+        try {
+          userSnippetsFileDisposable.add(userSnippetsFile.onDidChange(() => this.handleUserSnippetsDidChange()))
+          userSnippetsFileDisposable.add(userSnippetsFile.onDidDelete(() => this.handleUserSnippetsDidChange()))
+          userSnippetsFileDisposable.add(userSnippetsFile.onDidRename(() => this.handleUserSnippetsDidChange()))
+        } catch (e) {
+          const message = `\
+          Unable to watch path: \`snippets.cson\`. Make sure you have permissions
+          to the \`~/.atom\` directory and \`${userSnippetsPath}\`.
+
+          On linux there are currently problems with watch sizes. See
+          [this document][watches] for more info.
+          [watches]:https://github.com/atom/atom/blob/master/docs/build-instructions/linux.md#typeerror-unable-to-watch-path\
+          `
+          atom.notifications.addError(message, {dismissable: true})
+        }
+
+        callback(userSnippetsFileDisposable)
+      } else {
+        callback(new Disposable()) // TODO: Work out why it was passed -> in coffeescript
```

Comment:
The CoffeeScript code:
```coffeescript
foo = new Foo
```
Turns into the JavaScript:
```js
foo = new Foo;
```

While the CoffeeScript
```coffeescript
foo = new Foo ->
```
Turns into the JavaScript
```js
foo = new Foo(function() {});
```

----

So it looks like the correct translation here would be:
```js
callback(new Disposable(function () {}))
```

Reply:
That's what it was, but I thought it would be safe to remove the empty function. The disposable should have a working default construction ([and it appears to](https://github.com/atom/event-kit/blob/470fc3f1c433f5b848f7b3a2ae14474521dac422/lib/disposable.js#L35))

Reply:
Wow, somehow I completely glossed over the conditional and only saw the `this.disposalAction()` call when I was looking at that this morning 🤦‍♂️.